tests/backend/test_bosonic_backend.py

```diff
@@ -0,0 +1,560 @@
+    def test_cat_state_parity(self,alpha):
+        r"""Checks that the real and complex cat state representations
+        yield the correct parity."""    
+        # for phi = 0, should yield parity of 1
+        prog_complex_cat = sf.Program(1)
+        with prog_complex_cat.context as qc:
+            sf.ops.Catstate(alpha) | qc[0]
+        
+        prog_real_cat = sf.Program(1)
+        with prog_real_cat.context as qr:
+            sf.ops.Catstate(alpha,desc="real") | qr[0]
+        
+        backend_complex = bosonic.BosonicBackend()
+        backend_complex.run_prog(prog_complex_cat)
+        state_complex = backend_complex.state()
+        parity_complex = state_complex.parity_expectation([0])
+        
+        backend_real = bosonic.BosonicBackend()
+        backend_real.run_prog(prog_real_cat)
+        state_real = backend_real.state()
+        parity_real = state_real.parity_expectation([0])
+               
+        assert np.allclose(parity_complex, 1)
+        assert np.allclose(parity_real, 1)
+        
+        # for phi = 1, should yield parity of -1 unless alpha == 0
+        if alpha != 0:
+            prog_complex_cat = sf.Program(1)
+            with prog_complex_cat.context as qc:
+                sf.ops.Catstate(alpha,1) | qc[0]
+            
+            prog_real_cat = sf.Program(1)
+            with prog_real_cat.context as qr:
+                sf.ops.Catstate(alpha,1,desc="real") | qr[0]
+            
+            backend_complex = bosonic.BosonicBackend()
+            backend_complex.run_prog(prog_complex_cat)
+            state_complex = backend_complex.state()
+            parity_complex = state_complex.parity_expectation([0])
+            
+            backend_real = bosonic.BosonicBackend()
+            backend_real.run_prog(prog_real_cat)
+            state_real = backend_real.state()
+            parity_real = state_real.parity_expectation([0])
+                   
+            assert np.allclose(parity_complex, -1)
+            assert np.allclose(parity_real, -1)
+
+class TestBosonicFockStates:
+    r"""Tests fock state method of the BosonicBackend class."""
+    
+    @pytest.mark.parametrize("n", FOCK_VALS)
+    def test_fock(self,n):
+        r"""Checks fock states in the bosonic representation."""
+        prog = sf.Program(1)
+        with prog.context as q:
+            sf.ops.Fock(n) | q[0]
+              
+        backend = bosonic.BosonicBackend()
+        backend.run_prog(prog)
+        state = backend.state()
+        
+        #Check shapes
+        assert state.num_weights == n+1
+        assert state.weights().shape == (n+1,)
+        assert np.allclose(sum(state.weights()),1)
+        assert state.means().shape == (n+1,2)
+        assert state.covs().shape == (n+1,2,2)
+        
+        #Check mean photon is close to n
+        mean,var = state.mean_photon(0)
+        assert np.allclose(mean,n,atol=r_fock)
+        assert np.allclose(var,0,atol=r_fock)
```

Comment:
```suggestion
        assert np.allclose(mean, n, atol=r_fock)
        assert np.allclose(var, 0, atol=r_fock)
```

Reply:
There are a lot of missing spaces after `,` and between operations. Might be good to run a linter to make it a bit more legible?